pkg/platform/runtime/buildscript/file.go

```diff
@@ -0,0 +1,103 @@
+package buildscript
+
+import (
+	"errors"
+	"os"
+	"path/filepath"
+
+	"github.com/ActiveState/cli/internal/constants"
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/fileutils"
+	"github.com/ActiveState/cli/internal/logging"
+	"github.com/ActiveState/cli/pkg/platform/api/graphql/model/buildplanner"
+
+	"gopkg.in/yaml.v2"
+)
+
+type DoesNotExistError struct{ error }
+
+func IsDoesNotExistError(err error) bool {
+	return errs.Matches(err, &DoesNotExistError{})
+}
+
+type File struct {
+	Path   string
+	Script *Script
+}
+
+func newFile(path string) (*File, error) {
+	data, err := fileutils.ReadFile(path)
+	if err != nil {
+		if errors.Is(err, os.ErrNotExist) {
+			return nil, &DoesNotExistError{errs.New("Build script '%s' does not exist", path)}
+		}
+		return nil, errs.Wrap(err, "Could not read build script")
+	}
```

Comment:
I find it preferable to have disk ops isolated away from testable logic. In other words, letting disk operations be tested by tests that are more integration-focused. Consider moving this disk op into NewFile, and then changing newFile to be `newFile(data []byte) (*File, error)`. NewFile will probably then need to receive a full path, and the path construction would occur within UpdateOrCreate to be passed into both NewFile and file.update. This might also mean that the json test data would live in .go files as variables. Again, that would be more normal, in my opinion/experience, for type/package tests of lower level logic. The e2e tests would be the tests which rely on disk ops and json files.

Reply:
`newFile()` still needs a path in order to construct a `File{}`, so your suggestion will not work. However, I've changed the tests to use `NewScript()` instead, and inlined the buildscripts and buildexpressions as you suggested. Integration tests in future PRs will do e2e testing of `NewFile()`, so it can be revisited then if necessary.

Reply:
If I understand correctly, the path field would no longer need to be held by the File type. In other words, both the write and update functions can not be methods. It feels to me that this would work towards simplifying File itself. However, feel free to disregard this thread.

Reply:
Thanks for the clarification. I have removed the `File` struct in favor of using functions that operate directly on paths.